src/components/Nav/Navigation.tsx

```diff
@@ -106,25 +49,42 @@ class Navigation extends React.Component<PropsType, StateType> {
   }
 
   setControlledState = event => {
-    if (event.activePath) {
-      // keep track of path as user clicks on nav bar
-      this.setState({ selectedItem: event.activePath });
-    } else if ('navCollapsed' in event) {
+    if ('navCollapsed' in event) {
       this.props.setNavCollapsed(this.props.navCollapsed);
     }
   };
 
+  renderMenuItems() {
+    const { location } = this.props;
+    const menuItems = routes();
+    const activeItem = menuItems.find(item => {
+      let isRoute = matchPath(location.pathname, { path: item.to, exact: true, strict: false }) ? true : false;
+      if (!isRoute && item.pathsActive) {
+        isRoute = _.filter(item.pathsActive, path => path.test(location.pathname)).length > 0;
```

Comment:
loadsh is used because we don't know the type and we can filter correctly ?
Or what is the use case ?

Reply:
The case is know which menu item should be active so if path not match with the "normal paths" stored in each item.path in store I'll check if there is a pathsActive that match.

This is to have active services for example when the path is /\/namespaces\/(.*)\/services\/(.*)/g

Reply:
Changed